gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -914,22 +914,22 @@ def test_{{ method_name }}_rest(request_type):
                 result = value
 
             if result:
-                for nested_key in result.keys():
-                    if (key, nested_key) not in subfield_names:
-                        keys_to_delete.append(
-                            {"key": key, "nested_key": nested_key, "is_repeated": is_repeated}
+                for subfield in result.keys():
+                    if (field, subfield) not in nested_fields:
+                        subfields_to_delete.append(
+                            {"field": field, "subfield": subfield, "is_repeated": is_repeated}
                         )
 
         # Remove fields from the sample request which are not present in the runtime version of the dependency
-        for key_to_delete in keys_to_delete:
-            if key_to_delete.get("nested_key"):
-                if key_to_delete.get("is_repeated"):
-                    del request_init["{{ field.name }}"][key_to_delete.get("key")][0][
-                        key_to_delete.get("nested_key")
+        for subfield_to_delete in subfields_to_delete:
```

Comment:
nit: these two variables differ only by a letter, which seems brittle. Could we make them more distinct? One idea: s/`subfields_to_delete`/`subfields_not_in_runtime`/ or s/`subfields_to_delete`/`non_runtime_subfields`/

Reply:
Fixed in dc7934d5b347f5d98113befc97798ac5e4a9fb44